### PR TITLE
Updates Italian translations

### PR DESCRIPTION
**Pull request by bosskovic**
We updated Italian translations by adding missing key-value pairs from [en.yml](https://github.com/solidusio/solidus/blob/master/core/config/locales/en.yml) and translating the values to Italian. We did not remove any keys that were previously present, so it.yml now has more keys then en.yml of solidus core.

We tested the translations only by manually navigating the app and we made sure that no translations were missing by raising exceptions for missing translations (`config.action_view.raise_on_missing_translations = true`)

In addition to this we updated the Italian localization files for other solidus gems and will send pull requests to their repos.

### CODE REVIEW

**Comment by tvdeyen** [comment]
#64 fixes the build failures. Please rebase once it's landed in `master`

**Comment by bosskovic** [comment]
@tvdeyen what do I need to do in order to help this get into the master so that we can use the main gem instead of a fork?

**Comment by tvdeyen** [comment]
@bosskovic how is your work related to #56?

Please 

1. Rebase with current `master` now that #65 landed to fix the build errors
2. Stash all commits into one commit that is called "Updates Italian translation" or similar, because currently the commit messages doesn't provide any value (http://chris.beams.io/posts/git-commit/)

Thanks for your contribution

**Comment by dfranciosi** [comment]
@bosskovic
There are too many additions to comment line by line, so I post you direct links:

https://github.com/Xpitality/solidus_i18n/blob/e9070d890c1e5bfd4518814301f96b873fd6129f/config/locales/it.yml#L5
typo, should be `Quantità`

https://github.com/Xpitality/solidus_i18n/blob/e9070d890c1e5bfd4518814301f96b873fd6129f/config/locales/it.yml#L317
should be `Icona`

**Comment by mamhoff** [comment]
@bosskovic Would you mind squashing your commits and rebasing on master?

File: config/locales/it.yml [hunk 1]
```diff
@@ -1,51 +1,100 @@
 it:
+  activemodel:
+    attributes:
+      spree/order_cancellations:
+        quantity: Quantà
+        state: Status
```
Comment (alepore) [inline config/locales/it.yml]: i think we usually use "Stato" for state?

File: config/locales/it.yml [hunk 2]
```diff
@@ -1,51 +1,100 @@
...
+        shipment: Trasporto
+        cancel: Annulla
   activerecord:
     attributes:
       spree/address:
         address1: Indirizzo
         address2: Indirizzo (agg.)
         city: Città
-        country: Nazione
```
Comment (alepore) [inline config/locales/it.yml]: country must be deleted?

File: config/locales/it.yml [hunk 3]
```diff
@@ -1,51 +1,100 @@
...
+        company: Azienda
         firstname: Nome
         lastname: Cognome
         phone: Telefono
-        state: Provincia
```
Comment (alepore) [inline config/locales/it.yml]: deleted?

File: config/locales/it.yml [hunk 4]
```diff
@@ -1,51 +1,100 @@
 it:
+  activemodel:
+    attributes:
+      spree/order_cancellations:
+        quantity: Quantà
```
Comment (alepore) [inline config/locales/it.yml]: *Quantità

File: config/locales/it.yml [hunk 5]
```diff
@@ -1,51 +1,100 @@
...
+        state: Status
+        shipment: Trasporto
```
Comment (alepore) [inline config/locales/it.yml]: I think shipment is usually "Spedizione".
We ned to keep coherence between translations

File: config/locales/it.yml [hunk 6]
```diff
@@ -1,51 +1,100 @@
...
         lastname: Cognome
         phone: Telefono
-        state: Provincia
         zipcode: CAP
+      spree/adjustment:
+        adjustable: Modificabile
+        amount: Totale
+        label: Descrizione
+        name: Nome
+        state: Status
```
Comment (alepore) [inline config/locales/it.yml]: It's again "Status" here, but "Stato" a few lines below

File: config/locales/it.yml [hunk 7]
```diff
@@ -1,51 +1,100 @@
...
+        name: Nome
+        state: Stato
       spree/calculator/tiered_flat_rate:
         preferred_base_amount: Importo Base
         preferred_tiers: Livelli
       spree/calculator/tiered_percent:
         preferred_base_percent: Percentuale Base
         preferred_tiers: Livelli
+      spree/carton:
+        tracking: Tracking
```
Comment (alepore) [inline config/locales/it.yml]: "Tracciamento"?

File: config/locales/it.yml [hunk 8]
```diff
@@ -1,51 +1,100 @@
...
         name: Nome
         number: Numero
-        verification_value: Codice di Verifica
+        verification_value: Codice di Verifica (CVV)
         year: Anno
+      spree/customer_return:
+        number: Numero di restituzione
+        pre_tax_total: Totale (tasse escluse)
+        total: Totale
+        created_at: "Data/Ora"
```
Comment (alepore) [inline config/locales/it.yml]: I'd uniform all of this with "Creato il"

File: config/locales/it.yml [hunk 9]
```diff
@@ -1,51 +1,100 @@
...
+        additional_tax_total: Tax
+        approved_at: Approvato il
+        approver_id: Approvato da
+        canceled_at: Cancellato il
+        canceler_id: Cancellato da
         checkout_complete: Checkout Completato
         completed_at: Completato il
         considered_risky: A Rischio
-        coupon_code: Codice Coupon
+        coupon_code: Coupon
```
Comment (alepore) [inline config/locales/it.yml]: Doesn't seem an improvement to me

File: config/locales/it.yml [hunk 10]
```diff
@@ -1,51 +1,100 @@
...
         created_at: Data Ordine
         email: E-Mail Cliente
+        included_tax_total: Totale tasse (incl.)
```
Comment (alepore) [inline config/locales/it.yml]: Totale (tasse incl.)

File: config/locales/it.yml [hunk 11]
```diff
@@ -72,80 +121,218 @@ it:
...
+        is_default: Valido
       spree/product:
         available_on: Disponibile dal
         cost_currency: Valuta Costo
         cost_price: Prezzo di Costo
         description: Descrizione
+        depth: Profondità
+        height: Altezza
         master_price: Prezzo
+        meta_description: Meta Descrizione
```
Comment (alepore) [inline config/locales/it.yml]: I'd stay with "Meta description" here...

File: config/locales/it.yml [hunk 12]
```diff
@@ -72,80 +121,218 @@ it:
...
+        meta_keywords: Meta Keywords
+        meta_title: Meta Titolo
```
Comment (alepore) [inline config/locales/it.yml]: I'd stay with "Meta title"

File: config/locales/it.yml [hunk 13]
```diff
@@ -72,80 +121,218 @@ it:
...
+        active: Attivo
+        name: Nome
+        code: Codice
+      spree/reimbursement:
+        created_at: "Data/Ora"
+        number: Numero
+        reimbursement_status: Status
+        total: Totale
+      spree/reimbursement/credit:
+        amount: Totale
```
Comment (alepore) [inline config/locales/it.yml]: Amount is not usually "totale"...

File: config/locales/it.yml [hunk 14]
```diff
@@ -72,80 +121,218 @@ it:
...
+      spree/reimbursement_type:
+        name: Nome
+        type: Tipo
+        created_at: "Data/Ora"
       spree/return_authorization:
-        amount: Quantità
+        amount: Totale
```
Comment (alepore) [inline config/locales/it.yml]: mhhh

File: config/locales/it.yml [hunk 15]
```diff
@@ -72,80 +121,218 @@ it:
...
+        state_id: Provincia
+        zipcode: CAP
+      spree/stock_movement:
+        action: Azione
+        quantity: Quantità
+      spree/stock_transfer:
+        created_at: Creato il
+        created_by_id: Creato da
+        description: Descrizione
+        destination_location_id: Location di destionazione
```
Comment (alepore) [inline config/locales/it.yml]: *destinazione

File: config/locales/it.yml [hunk 16]
```diff
@@ -162,139 +349,176 @@ it:
...
         one: Indirizzo
         other: Indirizzi
+      spree/adjustment:
+        one: modifica
+        other: Modifiche
+      spree/adjustment_reason:
+        one: Motivo modifica
+        other: Motivi modifica
+      spree/calculator:
+        one: Calcolatrice
```
Comment (alepore) [inline config/locales/it.yml]: I think it's more "Calcolatore"

File: config/locales/it.yml [hunk 17]
```diff
@@ -162,139 +349,176 @@ it:
...
-        other: Unità di Inventario
+        one: Unità in inventario
+        other: Unità in inventario
+      spree/legacy_user:
+        one: Utente
+        other: Utenti
       spree/line_item:
-        one: Riga d'Ordine
-        other: Righe d'Ordine
+        one: elemento ordine
```
Comment (alepore) [inline config/locales/it.yml]: i think "riga d'ordine" is the common term for this in Italian...

File: config/locales/it.yml [hunk 18]
```diff
@@ -162,139 +349,176 @@ it:
...
-        other: Metodi di Spedizione
+        one: Metodo di spedizione
+        other: Metodi di spedizione
       spree/state:
         one: Provincia
         other: Province
       spree/state_change:
-        one: Cambio di Stato
-        other: Cambi di Stato
+        one: Cambio di status
```
Comment (alepore) [inline config/locales/it.yml]: 👎

File: config/locales/it.yml [hunk 19]
```diff
@@ -162,139 +349,176 @@ it:
...
+        other: Cambi di status
+      spree/stock: stock
+      spree/stock_item:
+        one: Articolo di stock
+        other: Articoli di stock
```
Comment (alepore) [inline config/locales/it.yml]: This doesn't sound right...

File: config/locales/it.yml [hunk 20]
```diff
@@ -316,6 +543,78 @@ it:
       spree/zone:
         one: Zona
         other: Zone
+    errors:
+      models:
+        spree/calculator/tiered_flat_rate:
+          attributes:
+            base:
+              keys_should_be_positive_number: "Le chiavi dei livelli devono essere numero maggiore di 0"
```
Comment (alepore) [inline config/locales/it.yml]: "un numero..." ?

File: config/locales/it.yml [hunk 21]
```diff
@@ -316,6 +543,78 @@ it:
...
+            preferred_tiers:
+              should_be_hash: "dovrebbe essere un hash"
+        spree/calculator/tiered_percent:
+          attributes:
+            base:
+              keys_should_be_positive_number: "Le chiavi dei livelli devono essere numero maggiore di 0"
```
Comment (alepore) [inline config/locales/it.yml]: as above

File: config/locales/it.yml [hunk 22]
```diff
@@ -316,6 +543,78 @@ it:
...
+            preferred_tiers:
+              should_be_hash: "dovrebbe essere un hash"
+        spree/classification:
+          attributes:
+            taxon_id:
+              already_linked: "è già collegato a questo prodotto"
+        spree/credit_card:
+          attributes:
+            base:
+              card_expired: "La carta è escaduta"
```
Comment (alepore) [inline config/locales/it.yml]: *scaduta

File: config/locales/it.yml [hunk 23]
```diff
@@ -316,6 +543,78 @@ it:
...
+              disallowed_with_code: Proibito per promozioni con un codice
+              disallowed_with_path: Proibito per promozioni con un path
+        spree/refund:
+          attributes:
+            amount:
+              greater_than_allowed: è maggiore dell'ammontare consentito.
+        spree/reimbursement:
+          attributes:
+            base:
+              return_items_order_id_does_not_match: One or more of the return items specified do not belong to the same order as the reimbursement.
```
Comment (alepore) [inline config/locales/it.yml]: not translated

File: config/locales/it.yml [hunk 24]
```diff
@@ -326,28 +625,28 @@ it:
...
-      failure: "Non è stato possibile autorizzarti da %{kind} perché %{reason}."
+      failure: "Non è stato possibile autorizzare il tuo account tramite %{kind}: %{reason}."
       success: "Autorizzato correttamente dall'account %{kind}"
     unlocks:
       send_instructions: Riceverai un'email con le istruzioni per sbloccare il tuo account entro pochi minuti.
       unlocked: Il tuo account è stato sbloccato con successo. Hai effettuato il login.
     user_passwords:
       user:
-        cannot_be_blank: La password non può essere vuota.
+        cannot_be_blank: Devi inserire una password valida.
```
Comment (alepore) [inline config/locales/it.yml]: 👎

File: config/locales/it.yml [hunk 25]
```diff
@@ -360,70 +659,105 @@ it:
...
-    acceptance_errors: Errori Accettazione
-    acceptance_status: Stato Accettazione
+    acceptance_errors: Errori nell'accettazione
+    acceptance_status: Stato dell'accettazione
     accepted: Accettato
     account: Account
     account_updated: Account aggiornato
     action: Azione
     actions:
+      add: Add
```
Comment (alepore) [inline config/locales/it.yml]: Aggiungi?

File: config/locales/it.yml [hunk 26]
```diff
@@ -360,70 +659,105 @@ it:
...
       cancel: Annulla
       continue: Continua
       create: Crea
+      delete: Delete
```
Comment (alepore) [inline config/locales/it.yml]: This is also "Elimina" in italian i think...

**Comment by bosskovic** [comment]
Hi!

It was easier for me to create a new pull request: https://github.com/solidusio-contrib/solidus_i18n/pull/90

I based it on the most recent master. I used the en.yml from solidus core as reference (added the missing keys and fixed the order of existing keys to match that in en.yml as much as possible). I took the values from our Italian translations and the existing it.yml of solidus_i18n repo.

In it.yml now there are lot more keys then in en.yml of solidus core. Possibly many are obsolete, but I did not remove any of them myself.

Please note that I am not an Italian speaker, so I did not (nor am I able to) add any new translations, I just used those that were already there and picked and chose between the two when there were variants (one being the existing it.yml translations and the other those translated by our team in the past year).

Please take a look and merge it soon, because it is quite time consuming doing these compares and resolving conflicts.

**Comment by bosskovic** [comment]
I'd like to emphasise again that I changed the order of keys in it.yml of solidus_i18n to match the order of keys in en.yml of solidus_core. Thus many keys will appear to be deleted in the diff, but if you look carefully, they are added back at some other place. it.yml now contains all keys from en.yml + some other (probably mostly obsolete).

**Comment by afdev82** [comment]
What's the status on this? I wanted to contribute, but I don't understand... I don't want to do the work twice

**Comment by coorasse** [comment]
Closed in favour of #90 and #138 . Thanks for your work @bosskovic !

**Pull request closed without merging**